.drone.yml

```diff
@@ -3,24 +3,26 @@ name: build image
 
 steps:
 - name: build image
-  image: vmck/vmck:0.0.2
+  image: hashicorp/packer:1.4.2
   privileged: true
   volumes:
   - name: vmck-images
-    path: /vmck-images
+    path: /drone/src/packer-build/vmck-images
   commands:
-  - IMAGE_NAME=imgbuild-$(echo $DRONE_COMMIT_BRANCH | tr -s '/' '_').qcow2
-  - SCRIPT=/drone/src/scripts/$(echo $DRONE_COMMIT_BRANCH | cut -f1 -d'/').sh
-  - ARCHIVE_TMP=/vmck-images/$IMAGE_NAME__tmp.tar.gz
-  - ARCHIVE=/vmck-images/$IMAGE_NAME.tar.gz
-  - env
-  - ls -al $SCRIPT
-  - mkdir /tmp/img
-  - /opt/vmck/contrib/build.py /tmp/$IMAGE_NAME --script $SCRIPT
-  - tar czvf $ARCHIVE_TMP -C /tmp $IMAGE_NAME
+  - apk add --update libvirt-daemon qemu qemu-system-x86_64 qemu-img sshpass openssh
+  - addgroup $(whoami) libvirt
+  - export PACKER_LOG=1
+  - export IMAGE_NAME=imgbuild-"$(echo $DRONE_COMMIT_BRANCH | tr -s '/' '_')".qcow2
+  - export ARCHIVE_TMP=packer-build/build/"$IMAGE_NAME"__tmp.tar.gz
+  - export ARCHIVE=packer-build/vmck-images/"$IMAGE_NAME".tar.gz
+  - ./packer-build/build.sh --vm-name $IMAGE_NAME ubuntu1804.json
+  - ./packer-build/start.sh $IMAGE_NAME --detached --ssh 6622
+  - sleep 10
```

Comment:
This sleep is deeply suspicious. Are you waiting for something to finish? Is 10 seconds always enough? Is it just two seconds sometimes and we're wasting time? So many questions for a casual reader. A nicer pattern is to wait for a specific condition, with a timeout, like in other places. An even better pattern is to not need it at all.

Reply:
I use it to wait for the machine to boot and have ssh up and running. Qemu uses this constant when it waits for the machine to boot. I don't know of a way of checking if ssh is available on a remote machine without getting a non-zero return code.

Reply:
Seems like `until` does the job.